packages/flutter/lib/src/cupertino/form_row.dart

```diff
@@ -0,0 +1,198 @@
+  /// row. Standard iOS guidelines encourage passing a [Text] widget to [prefix]
+  /// to detail the nature of the row's [child] widget.
+  ///
+  /// The [margins] parameter is used to pad the contents of the row. It defaults
+  /// to the standard iOS padding. If no edge insets are intended, explicitly
+  /// pass [EdgeInsets.zero] to [margins].
+  ///
+  /// The [helper] and [error] parameters are both widgets targeted at displaying
+  /// more information about the row. Both widgets are placed underneath the
+  /// [prefix] and [child]. When a [Text] is given to [error], it will be shown in
+  /// [CupertinoColors.destructiveRed] coloring and medium-weighted font.
+  const CupertinoFormRow({
+    Key? key,
+    required this.child,
+    this.prefix,
+    this.margins,
+    this.helper,
+    this.error,
+  }) : super(key: key);
+
+  /// A widget that is displayed at the start of the row.
+  ///
+  /// The [prefix] parameter is displayed at the start of the row. Standard iOS
+  /// guidelines encourage passing a [Text] widget to [prefix] to detail the
+  /// nature of the row's [child] widget.
+  final Widget? prefix;
+
+  /// Margins used to pad the contents of the row.
+  ///
+  /// Defaults to the standard iOS padding for form rows. If no edge insets are
+  /// intended, explicitly pass [EdgeInsets.zero] to [margins].
+  final EdgeInsetsGeometry? margins;
+
+  /// A widget that is displayed underneath the [prefix] and [child] widgets.
+  ///
+  /// The [helper] appears in primary label coloring, and is meant to inform the
+  /// user about interaction with the child widget.
+  final Widget? helper;
+
+  /// A widget that is displayed underneath the [prefix] and [child] widgets.
+  ///
+  /// The [error] widget is primarily used to inform users of input errors. When
+  /// a [Text] is given to [error], it will be shown in
+  /// [CupertinoColors.destructiveRed] coloring and medium-weighted font.
+  final Widget? error;
+
+  /// Child widget.
+  ///
+  /// The [child] widget is primarily used for input. It end-aligned and
+  /// flexible, taking up the entire space trailing past the [prefix] widget.
+  final Widget child;
+
+  @override
+  Widget build(BuildContext context) {
+    final CupertinoThemeData themeData = CupertinoTheme.of(context);
+    final TextStyle textStyle = themeData.textTheme.textStyle;
+
+    final List<Widget> rowChildren = <Widget>[
+      if (prefix != null)
+        DefaultTextStyle(
+          style: textStyle,
+          child: prefix!,
+        ),
+      Flexible(
+        child: Align(
+          alignment: AlignmentDirectional.centerEnd,
+          child: child,
+        ),
+      ),
+    ];
+
+    return DecoratedBox(
+      decoration: BoxDecoration(
+        color: CupertinoColors.systemBackground.resolveFrom(context),
+      ),
+      child: Padding(
+        padding: margins ?? _kDefaultMargins,
+        child: Column(
+          mainAxisAlignment: MainAxisAlignment.start,
```

Comment:
do you need both this and the aligns?

Reply:
i need the aligns, so i can gut this line.